tensorflow/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -46,6 +46,9 @@ limitations under the License.
 #include "tensorflow/stream_executor/stream_executor_pimpl.h"
 // clang-format off
 #include "third_party/gpus/cudnn/cudnn.h"
+#if CUDNN_VERSION >= 8100
+#include "third_party/cudnn_frontend/include/cudnn_frontend.h"
```

Comment:
Overall comment: this PR is too large. Can you split it into multiple PRs, or at least multiple commits? I can see clear layering here, e.g. stream_executor doesn't depending on core/kernels.